src/hpa.c

```diff
@@ -712,14 +715,22 @@ hpa_from_pai(pai_t *self) {
 
 static size_t
 hpa_alloc_batch(tsdn_t *tsdn, pai_t *self, size_t size, size_t nallocs,
-    edata_list_active_t *results, bool *deferred_work_generated) {
+    edata_list_active_t *results, bool frequent_reuse,
+    bool *deferred_work_generated) {
 	assert(nallocs > 0);
 	assert((size & PAGE_MASK) == 0);
 	witness_assert_depth_to_rank(tsdn_witness_tsdp_get(tsdn),
 	    WITNESS_RANK_CORE, 0);
 	hpa_shard_t *shard = hpa_from_pai(self);
 
-	if (size > shard->opts.slab_max_alloc) {
+	/*
+	 * Allow frequent reused alloc to bypass the slab_max_alloc limit, as
```

Comment:
Should we use a more informative parameter name than `frequent_reuse`? In and of itself, a size being "frequently reused" does not imply no internal fragmentation; here the information we're conveying is that this request comes from the bins so there is no intrinsic slack in the allocation, because it is sized accurately.

So maybe something like `exact_size` or `no_slack`?

Reply:
Agreed the `frequent_reuse` is rather not explicit. However it's already passed down to the page allocator level (the page alloc interface's alloc function like hpa_alloc added it for a different reason). So renaming only here would make it a bit inconsistent.

Given that it's only necessary until we have the full size support (the new size classing strategy by Guangli), I'd say keep the current naming for now, but let me add more details in the comments.